api/internal/handler/route/route.go

```diff
@@ -392,16 +398,14 @@ func (h *Handler) Update(c droplet.Context) (interface{}, error) {
 		script := &entity.Script{}
 		script.ID = input.ID
 		script.Script = input.Script
-		//to lua
+		// Explicitly to lua if input script is of the map type, otherwise
+		// it will always represent a piece of lua code of the string type.
 		var err error
-		scriptConf, ok := input.Script.(map[string]interface{})
-		if !ok {
-			return &data.SpecCodeResponse{StatusCode: http.StatusBadRequest},
-				fmt.Errorf("invalid `script`")
-		}
-		input.Route.Script, err = generateLuaCode(scriptConf)
-		if err != nil {
-			return &data.SpecCodeResponse{StatusCode: http.StatusInternalServerError}, err
+		if scriptConf, ok := input.Script.(map[string]interface{}); ok {
+			input.Route.Script, err = generateLuaCode(scriptConf)
+			if err != nil {
+				return &data.SpecCodeResponse{StatusCode: http.StatusInternalServerError}, err
+			}
```

Comment:
Hi, @imjoey, we had `loadstring` in apisix to validate the script.
Can we do same things in dashboard?

Reply:
@starsz thanks for your advice. PR renewed and an additional validation check was added. Looking forward to your feedback. Much appreciated.

Reply:
Well done! Thanks a lot.